main.go

```diff
@@ -154,12 +154,39 @@ func run(ctx context.Context) {
 func runLoop(ctx context.Context) {
 	var taskNotificationClient *agentendpoint.Client
 	var err error
-	ticker := time.NewTicker(config.SvcPollInterval())
-	for {
-		if err := config.SetConfig(ctx); err != nil {
-			logger.Errorf(err.Error())
+
+	go func() {
+		for {
+			if err := config.WatchConfig(ctx, func() {
```

Comment:
Remove the func as WatchConfig returns, just have the logic run before WatchConfig here as we already have config setup at this point, then it will just run every time WatchConfig returns

Reply:
done